controller/appcontroller.go

```diff
@@ -2433,6 +2434,13 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 				if ctrl.hydrator != nil {
 					ctrl.appHydrateQueue.AddRateLimited(newApp.QualifiedName())
 				}
+				// if cluster has been changed, then let's clean resources of the oldApp.
+				if newOK && oldOK && oldApp.Spec.Destination.Server != newApp.Spec.Destination.Server {
+					if err := ctrl.deleteAppResources(oldApp); err != nil {
```

Comment:
I am thinking what if deletion fails?
Is there an easy way how to queue it and retry?